### ISSUE TITLE
Large bulk inserts using db.bulk_docs() results in a timeout

### ISSUE THREAD

**Issue by illeatmyhat** [cloudant/python-cloudant#331]
Cloudant version 2.6.0
Python version 3.6.2

In the interest of being able to insert and update large datasets because of API call limits, I use db.bulk_docs() to insert a large amount of documents.

If we attempt to use db.bulk_docs() to insert a large amount of data, we get a timeout.
The workaround is to simply insert less documents at a time, though not knowing an exact limitation is rather painful.

```
data = [{"very-large-json-objects":"almost 2MB combined"}]
db.bulk_docs(data)
```
This will result in `requests.exceptions.ConnectionError: ('Connection aborted.', BrokenPipeError(32, 'Broken pipe'))`

After some stepping, it seems that the POST request is sent using unchunked transfer encoding.
Perhaps consider giving the option to use chunked?

**Comment by ricellis** [comment]
>In the interest of being able to insert and update large datasets because of API call limits, I use db.bulk_docs()

FWIW each document in a `_bulk_docs` write will count towards the API write limits (I think that was omitted initially) so you'll ultimately need a plan with an appropriate number of writes anyway. That said bulk transfers are more efficient - so you're doing the correct thing to use bulk for mass inserts. 

However, there is a HTTP request size limit on the server, which I believe is currently 1 MB. Given your 2 MB data it seems more likely that size limit is the cause of your `Connection aborted`, rather than a timeout. The server does return a `413` before aborting the connection, but we have noticed some HTTP libraries don't read data from the server until they've finished sending and so fail with broken connections instead of the correct error code. It [looks like requests](https://github.com/requests/requests/issues/2422#issuecomment-258370446) is another one of these libraries.

**Comment by illeatmyhat** [comment]
> However, there is a HTTP request size limit on the server, which I believe is currently 1 MB.

That makes sense. It seems like requests will not be able to retrieve the proper 413 error anytime soon. To work around that, since this is a Cloudant specific setting, it would be helpful if ` _bulk_docs` would return a more intelligible response, or throw an exception, if the request body exceeds the request size limit.

That way people aren't thrown for a loop by unexpected broken pipes.

**Comment by ricellis** [comment]
Yeah I agree, that's why I left the ticket open. There might not be much we can do given how deep in the stack it is, but we'll look into it.

**Comment by tomblench** [comment]
This is the sort of issue that `Expect: 100-continue` might help with but it's not currently supported on the service.

**Comment by ricellis** [comment]
This is fixed in urllib3 1.26 which is usable by requests 2.25
this is in range for our requirements.txt so hopefully should get picked up with dependency upgrades.
